### PR TITLE
no modal for empty tree-group deletion

### LINKED ISSUE #4499: Deleting empty tree group should not prompt confirmation message
If you delete an empty tree group you still get asked if you want to delete its children. This step can be skipped in this case.

### PR DESCRIPTION
**Pull request by youri-k**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- delete empty tree group => should just work
- delete non-empty tree group => should still ask for confirmation

### Issues:
- fixes #4499 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/view/right-menu/trees_tab_view.js [hunk 1]
```diff
@@ -322,7 +322,19 @@ class TreesTabView extends React.PureComponent<Props, State> {
   };
 
   showDeleteGroupModal = (id: number) => {
-    this.setState({ groupToDelete: id });
+    if (!this.props.skeletonTracing) return;
+
+    const { trees, treeGroups } = this.props.skeletonTracing;
+    const treeGroupToDelete = treeGroups.find(el => el.groupId === id);
+    const groupToTreesMap = createGroupToTreesMap(trees);
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/right-menu/trees_tab_view.js]: Did you test to delete a group which only contains another group (which contains a tree)? Like this:
```
Group A
  Group B
    Tree 1
```

In that case, a modal should appear. If it doesn't work right now, you might want to use a helper function like `callDeep` or similar (have a look at `frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view_helpers.js`).

Comment (youri-k) [reply frontend/javascripts/oxalis/view/right-menu/trees_tab_view.js]: As I'm checking whether the group has a child group, I'm guarding against that case, but is this too strict?

Comment (youri-k) [reply frontend/javascripts/oxalis/view/right-menu/trees_tab_view.js]: So should it be allowed to delete two empty groups like:
```
Group A
  Group B
```

Comment (philippotto) [reply frontend/javascripts/oxalis/view/right-menu/trees_tab_view.js]: > As I'm checking whether the group has a child group, I'm guarding against that case

Ah, sorry, my bad! I missed that.

> So should it be allowed to delete two empty groups like:
> 
> ```
> Group A
>   Group B
> ```

I think, that's not necessary :)

**Comment by youri-k** [comment]
Ok perfect :slightly_smiling_face:

**Pull request merged**